src/main/webapp/js/new_simulation.js

```diff
@@ -36,7 +36,36 @@ function updateDatalistValuesFromArray(datalistId, valuesArray) {
     document.getElementById(datalistId).innerHTML = valuesHtml;
 }
 
-// Inputs' Dynamic Upper Boundes
+// Inputs' Dynamic Bounds
+
+/** Verify that the values of all input tags are legal, change illegal values. */
+function enforceRigidBounds() {
+    var roundsNum = document.getElementById('roundsNum').value;
+    document.getElementById('roundsNum').value = Math.max(1, roundsNum);
+
+    var beaconsNum = document.getElementById('beaconsNum').value;
+    document.getElementById('beaconsNum').value = Math.max(1, beaconsNum);
+
+    var observersNum = document.getElementById('observersNum').value;
+    document.getElementById('observersNum').value = Math.max(1, observersNum);
+
+    var currentRows = document.getElementById('rowsNum').value;
+    var maxRows = document.getElementById('rowsNum').max;
+    document.getElementById('rowsNum').value = Math.max(1, Math.min(maxRows, currentRows));
```

Comment:
Consider implementing a `setStaticUpperBound(fieldId, value)` function instead of repeating this code (in addition to the existing `setDynamicUpperBound()` function).